cylc/flow/scripts/lint.py

```diff
@@ -14,19 +14,18 @@
 #
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Cylc configuration linter.
+
+Checks code style, deprecated syntax and other issues."""
+# NOTE: docstring needed for `cylc help all` output
+# (if editing check this still comes out as expected)
+
 COP_DOC = """cylc lint [OPTIONS] ARGS
 
 Cylc lint looks through one or more folders for ".cylc" and ".rc"
 files.
 
-By default it will examine files for code style.
-If a workflow has been upgraded to Cylc 8 by changing the top level
-"suite.rc" to "flow.cylc" then it will also check for deprecated
-Cylc 7 syntax.
-
-You can run "cylc lint" with each set of rules alone using
-"cylc lint -r style" (for the style checks) or  "cylc lint -r 728"
-(for the Cylc 7 syntax checks).
```

Comment:
I don't think turning off rulesets is a large use case so the option docs should be enough.